lib/TestRunner.lua

```diff
@@ -53,51 +102,36 @@ function TestRunner.runPlanNode(session, planNode, tryStack, noXpcall)
 				childResultNode.status = TestEnum.TestStatus.Skipped
 			else
 				if tryStack:size() > 0 and tryStack:getBack().isOk == false then
+
 					childResultNode.status = TestEnum.TestStatus.Failure
 					table.insert(childResultNode.errors,
 						string.format("%q failed without trying, because test case %q failed",
 							childPlanNode.phrase, tryStack:getBack().failedNode.phrase))
 				else
 					-- Errors can be set either via `error` propagating upwards or
 					-- by a test calling fail([message]).
-					local success = true
-					local errorMessage
-
-					local testEnvironment = getfenv(childPlanNode.callback)
+					local success, errorMessage = true, nil
 
-					for key, value in pairs(TestRunner.environment) do
-						testEnvironment[key] = value
+					for _, hook in pairs(lifecycleHooks:getPendingBeforeHooks()) do
+						if success then
+							success, errorMessage = runCallback(hook, false, "beforeAll hook: ")
+						end
```

Comment:
We don't run `afterAll` hooks in this block, but we run `beforeAll` here.

Why don't we process `beforeAll` above, before we enter this child loop?

Reply:
We can do that, but if the beforeAll node fails I'm afraid I wont have access to the child nodes it was wrapping as easily

Reply:
Also if we move beforeAll out, we wont be able to filter out describe blocks with beforeAlls with no It blocks very easily.

Reply:
Do we have the same problem with `afterAll` not being filtered? I wonder if we should try to patch this up a little in TestEZ and then start fresh in the new test framework with lifecycle events in mind from the start.